### ISSUE TITLE
Not all of generators can be translated into ordinal functions

### ISSUE THREAD

**Issue by vadzim** [babel/babel#1545]
I reproduced this bug on clean machine with debian, but I cannot tell for sure which version of babel and in which environment one must have to reproduce the bug.
Babel v5.3.3 has a bug on one machine, but do not has on another.
Babel v5.4.2 has a bug on three machines I have tested on.

Bug:

First generator function is converted into ordinal function, but the second one isn't.

The output is:
```
# babel
function*a(){}
function*b(){}
"use strict";

var marked0$0 = [a].map(regeneratorRuntime.mark);
function a() {
  return regeneratorRuntime.wrap(function a$(context$1$0) {
    while (1) switch (context$1$0.prev = context$1$0.next) {
      case 0:
      case "end":
        return context$1$0.stop();
    }
  }, marked0$0[0], this);
}
function* b() {}

# node --version
v0.12.3
# babel --version
5.4.2
#
```
I've uploaded virtual machine image to [google drive](https://drive.google.com/file/d/0B3dzh6jVRzEOMVZwQnJPcG1uYkU/view?pli=1) to one to be able to reproduce the bug for sure. The size of the archive is about 370M. Sorry for that, but I've failed to start node.js on smaller vdi linux images I've found on the Internet.
username/password: deb/deb

**Comment by vadzim** [comment]
If source code is changed to smth like
```javascript
let a=function*a(){}           
let b=function*b(){}
```
then the output is okay:
```
"use strict";

var a = regeneratorRuntime.mark(function a() {
  return regeneratorRuntime.wrap(function a$(context$1$0) {
    while (1) switch (context$1$0.prev = context$1$0.next) {
      case 0:
      case "end":
        return context$1$0.stop();
    }
  }, a, this);
});
var b = regeneratorRuntime.mark(function b() {
  return regeneratorRuntime.wrap(function b$(context$1$0) {
    while (1) switch (context$1$0.prev = context$1$0.next) {
      case 0:
      case "end":
        return context$1$0.stop();
    }
  }, b, this);
});
```

**Comment by sebmck** [comment]
Regression due to #1536.

**Comment by NickHeiner** [comment]
:+1: I have the same issue – thanks for reporting it.

**Comment by sebmck** [comment]
Fixed as of 5.4.3, thanks!
